storagemarket/deal_execution.go

```diff
@@ -502,21 +530,23 @@ func (p *Provider) indexAndAnnounce(ctx context.Context, pub event.Emitter, deal
 	p.dealLogger.Infow(deal.DealUuid, "deal successfully added to piecestore")
 
 	// register with dagstore
-	if err := stores.RegisterShardSync(ctx, p.dagst, pc, deal.InboundFilePath, true); err != nil && !xerrors.Is(err, dagstore.ErrShardExists) {
+	err := stores.RegisterShardSync(ctx, p.dagst, pc, "", true)
+
+	if err != nil && !xerrors.Is(err, dagstore.ErrShardExists) {
 		return fmt.Errorf("failed to register deal with dagstore: %w", err)
+	} else if err != nil && xerrors.Is(err, dagstore.ErrShardExists) {
+		p.dealLogger.Infow(deal.DealUuid, "deal has previously been registered in dagstore")
+	} else {
+		p.dealLogger.Infow(deal.DealUuid, "deal has successfully been registered in the dagstore")
```

Comment:
nit: it would be easier for me to follow the logic if it were:
```suggestion
	if err != nil {
	        if !xerrors.Is(err, dagstore.ErrShardExists) {
    		    return fmt.Errorf("failed to register deal with dagstore: %w", err)
		}
		p.dealLogger.Infow(deal.DealUuid, "deal has previously been registered in dagstore")
	} else {
		p.dealLogger.Infow(deal.DealUuid, "deal has successfully been registered in the dagstore")
```